### PR TITLE
chore(deps): update kotlin and AGP version

### PR DESCRIPTION
**Pull request by duyhungtnn**
# Why do we need this PR 
- The Renova bot fails to increase SDK's deps  in this PR https://github.com/bucketeer-io/android-client-sdk/pull/52

# Changes 
- [x] Increase Kotlin version to `1.8.22`
- [x] Increase `com.google.devtools.ksp` to `1.8.22-1.0.11`
- [x] Increase `org.jetbrains.dokka` to `1.8.20`
- [x] Increase Android Gradle Plugin to `7.4.2` for compatibility with new Kotlin versions  
- [x] Update build.gradle files for compatibility with AGP 8.x.x

### CODE REVIEW

File: sample/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,15 +1,14 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="io.bucketeer.sample"
```
Comment (duyhungtnn) [inline sample/src/main/AndroidManifest.xml:3]: Using `package="io.bucketeer.sample" in `AndroidManifest.xml` is no longer recommended after AGP 8.0.0

File: sample/build.gradle [hunk 2]
```diff
@@ -43,6 +44,17 @@ android {
       signingConfig signingConfigs.release
     }
   }
+  buildFeatures {
+    buildConfig true
+  }
+  compileOptions {
+    sourceCompatibility JavaVersion.VERSION_17
```
Comment (duyhungtnn) [inline sample/build.gradle:51]: required by AGP 8.0.0 and the new Kotlin version

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 [versions]
-agp = "7.3.1"
+agp = "8.0.0"
```
Comment (duyhungtnn) [inline gradle/libs.versions.toml:2]: AGP 8.0.0 and its Kotlin Plugin started to support Kotlin version 1.8.x

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 grpc = "1.47.0"
-kotlin = "1.7.22"
+kotlin = "1.8.22"
 espresso = "3.4.0"
 androidxLifecycle = "2.5.1"
 androidxTest = "1.5.0"
 moshi = "1.14.0"
-okhttp = "4.10.0"
+okhttp = "4.11.0"
```
Comment (duyhungtnn) [inline gradle/libs.versions.toml:9]: okhttp = "4.11.0"  version fixed the build release problem with R8. 
https://github.com/square/okhttp/issues/7777

File: sample/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -1,15 +1,14 @@
...
     android:versionCode="1"
     android:versionName="1.0">
 
     <application
-        android:name=".App"
+        android:name="io.bucketeer.sample.App"
```
Comment (duyhungtnn) [inline sample/src/main/AndroidManifest.xml:7]: `.App` is no longer supported

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -16,7 +16,7 @@ jobs:
         uses: actions/setup-java@v3.11.0
         with:
           distribution: 'temurin'
-          java-version: 11
+          java-version: 17
```
Comment (duyhungtnn) [inline .github/workflows/build.yml:19]: JVM 17 is required by AGP 8.0.0 and the new Kotlin version

File: gradle/libs.versions.toml [hunk 7]
```diff
@@ -48,7 +48,7 @@ android-application = { id = "com.android.application", version.ref = "agp" }
 kotlin-android = { id = "org.jetbrains.kotlin.android", version.ref = "kotlin" }
 kotlin-jvm = { id = "org.jetbrains.kotlin.jvm", version.ref = "kotlin" }
 kotlin-kapt = { id = "org.jetbrains.kotlin.kapt", version.ref = "kotlin" }
-kotlin-dokka = { id = "org.jetbrains.dokka", version = "1.7.20" }
+kotlin-dokka = { id = "org.jetbrains.dokka", version = "1.8.20" }
```
Comment (duyhungtnn) [inline gradle/libs.versions.toml:51]: dokka 1.8.20 is supported Kotlin 1.8.x

File: gradle.properties [hunk 8]
```diff
@@ -4,6 +4,7 @@ android.testMinSdkVersion=21
 android.minSdkVersion=21
 android.targetSdkVersion=30
 android.useAndroidX=true
+android.defaults.buildfeatures.buildconfig=true
```
Comment (duyhungtnn) [inline gradle.properties:7]: AGP 8.x.x disable `BuildConfig` by default

**Comment by duyhungtnn** [comment]
@cre8ivejp Hi please help me to review this PR. Thank you.

File: gradle/libs.versions.toml (hunk 3 above)
Comment (yshrsmz) [reply gradle/libs.versions.toml:2]: Could you use 8.0.2 instead?

Comment (duyhungtnn) [reply gradle/libs.versions.toml:2]: @yshrsmz sure, let me try this path release 8.0.2

Comment (duyhungtnn) [reply gradle/libs.versions.toml]: @yshrsmz it's okay now.  We are using 8.0.2

File: bucketeer/build.gradle [hunk 9]
```diff
@@ -48,12 +48,12 @@ android {
   }
 
   compileOptions {
-    sourceCompatibility JavaVersion.VERSION_1_8
-    targetCompatibility JavaVersion.VERSION_1_8
+    sourceCompatibility JavaVersion.VERSION_17
+    targetCompatibility JavaVersion.VERSION_17
```
Comment (yshrsmz) [inline bucketeer/build.gradle:52]: Should we keep `VERSION_1_8` here, as this effectively requires SDK consumers to use AGP 8 or greater?

Comment (duyhungtnn) [reply bucketeer/build.gradle:52]: Hi @yshrsmz , I tested with an example project, this PR will make breaking changes with A
I found another way to deal with it. I will update the PR soon.

File: .github/workflows/build.yml (hunk 6 above)
Comment (duyhungtnn) [inline .github/workflows/build.yml:19]: Supported AGP 8. x.x in the near future

File: bucketeer/build.gradle (hunk 9 above)
Comment (duyhungtnn) [reply bucketeer/build.gradle]: Hi @yshrsmz, JVM target may not cause many problems. I tested with some old projects.
But AGP 8.x.x will 
- Make breaking changes for some projects using the legacy library for example `Butterknife` 
- The SDK's consumer needs to use Android Studio FlamingGo (This IDE will create many errors with their project) 

After checking, AGP 7.4.2 is the best choice. (which supports Kotlin 1.8.22).
Other changes in this PR are to fix an error and make our project work well with Android Flamingo.

Comment (yshrsmz) [reply bucketeer/build.gradle]: Technically, you should be able to use AGP 8.0.x while keeping source/target compatibility to java11.

AGP 8 requires java17, but it is just for running itself. You can use java17 to compile java11-compatible binary.

But I've tested locally and found one ksp problem: https://github.com/google/ksp/issues/1288

Comment (yshrsmz) [reply bucketeer/build.gradle]:
```gradle
kotlin {
  jvmToolchain(8)
}

android {
    compileOptions {
    sourceCompatibility JavaVersion.VERSION_1_8
    targetCompatibility JavaVersion.VERSION_1_8
  }

  kotlinOptions {
    jvmTarget = "1.8"
  }
}
```

This seems to work for me(at least test & assemble works)

Comment (duyhungtnn) [reply bucketeer/build.gradle]: I tried `jvmToolchain` solution before, and it works.

I think we should go with `7.4.2`.  
It just like the same with `7.3.1` but support the new Kotlin Gradle Plugin version for Kotlin `1.8.22`. The impact is the smallest.
We should wait a little time before jumping for full support AGP 8.x.x 
What do you think @yshrsmz ?

**Review by yshrsmz — APPROVED** [review]
👍

**Pull request merged**